lib/xdrgen/generators/python.rb

```diff
@@ -0,0 +1,666 @@
+          out.indent(2) do
+            encode_member typedef, out
+          end
+
+          out.puts "@classmethod"
+          out.puts "def unpack(cls, unpacker: Unpacker) -> \"#{typedef_name}\":"
+          out.indent(2) do
+            decode_member typedef, out
+            out.puts "return cls(#{typedef_name_underscore})"
+          end
+          render_xdr_utils(out, typedef_name)
+          out.puts <<~HEREDOC
+            def __eq__(self, other: object):
+                if not isinstance(other, self.__class__):
+                    return NotImplemented
+                return self.#{typedef_name_underscore} == other.#{typedef_name_underscore}
+
+            def __str__(self):
+                return f"<#{typedef_name} [#{typedef_name_underscore}={self.#{typedef_name_underscore}}]>"
+          HEREDOC
+        end
+        out.close
+      end
+
+      def render_import(out, member, container_name)
+        member_type = type_string member.type
+        unless is_base_type member.type or container_name == member_type
+          out.puts "from .#{member_type.underscore} import #{member_type}"
+        end
+      end
+
+      def render_union(union)
+        union_name = name union
+        union_name_underscore = union_name.underscore
+        @init_out.puts "from .#{union_name_underscore} import #{union_name}"
+
+        file_name = "#{union_name_underscore}.py"
+        out = @output.open(file_name)
+        render_common_import out
+
+        render_import out, union.discriminant, union_name
+
+        union.arms.each do |arm|
+          next if arm.void?
+          # This may cause duplicate imports, we can remove it with autoflake
+          render_import out, arm.declaration, union_name
+        end
+
+        out.puts "__all__ = ['#{union_name}']"
+
+        out.puts "class #{union_name}:"
+        out.indent(2) do
+          render_source_comment(out, union)
+          union_discriminant_name_underscore = union.discriminant.name.underscore
+          out.puts <<~HEREDOC
+            def __init__(
+                self,
+                #{union_discriminant_name_underscore}: #{type_hint_string union.discriminant, union_name},
+          HEREDOC
+
+          out.indent(2) do
+            union.arms.each do |arm|
+              next if arm.void?
+              out.puts "#{arm.name.underscore}: #{type_hint_string arm.declaration, union_name} = None,"
+            end
+          end
+
+          out.puts ") -> None:"
+          out.indent(2) do
+            union.arms.each do |arm|
+              next if arm.void?
+              render_array_length_checker arm, out
+            end
+
+            out.puts "self.#{union_discriminant_name_underscore} = #{union_discriminant_name_underscore}"
+            union.arms.each do |arm|
+              next if arm.void?
+              arm_name_underscore = arm.name.underscore
+              out.puts "self.#{arm_name_underscore} = #{arm_name_underscore}"
```

Comment:
These few lines appear to work only on arms, and not arm cases, which means cases that are grouped together will produce only a single line here. Is that as intended?

Reply:
I think it's working fine, it is used to generate the following code:

https://github.com/StellarCN/py-stellar-base/blob/560519d2821746ba6ae6164bcf5a471744066cbe/stellar_sdk/xdr/ledger_entry_change.py#L39-L42

Reply:
Yes that looks good. Also, TransactionResultResult also looks good and is a more complex union definition. https://github.com/StellarCN/py-stellar-base/blob/560519d2821746ba6ae6164bcf5a471744066cbe/stellar_sdk/xdr/transaction_result_result.py